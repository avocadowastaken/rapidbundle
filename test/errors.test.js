"use strict";

const fs = require("fs");
const path = require("path");

const execCLI = require("./utils/execCLI");

const FIXTURES_DIR = path.join(__dirname, "__fixtures__");
const ERROR_FIXTURES_DIR = path.join(FIXTURES_DIR, "errors");

for (const fixture of fs.readdirSync(ERROR_FIXTURES_DIR)) {
  const fixtureDir = path.join(ERROR_FIXTURES_DIR, fixture);
  if (!fs.statSync(fixtureDir).isDirectory()) continue;

  test(fixture, async () => {
    const [stdout, stderr, exitCode] = await execCLI(fixtureDir);

    expect(stderr).toBe("");
    expect(stdout).toMatchSnapshot();
    expect(exitCode).toBe(1);

    const distDir = path.join(fixtureDir, "dist");
    expect(fs.existsSync(distDir)).toBe(false);
  });
}
