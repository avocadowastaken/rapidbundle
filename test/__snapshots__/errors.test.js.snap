// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`invalid-action 1`] = `
[HH:mm:ss] Resolving entry point [started]
[HH:mm:ss] → Resolving from: <rootDir>/test/__fixtures__/errors/invalid-action/src
[HH:mm:ss] → Resolved: <rootDir>/test/__fixtures__/errors/invalid-action/src/index.ts
[HH:mm:ss] Resolving entry point [completed]
[HH:mm:ss] Creating GitHub action bundle [started]
[HH:mm:ss] Creating GitHub action bundle [failed]
[HH:mm:ss] → Invalid 'action.yml':
  .runs.main: Expected a string (got undefined)
  .runs.using: Expected a valid enumeration value (got an empty string)
`;

exports[`invalid-entry 1`] = `
[HH:mm:ss] Resolving entry point [started]
[HH:mm:ss] → Resolving from: <rootDir>/test/__fixtures__/errors/invalid-entry/src
[HH:mm:ss] Resolving entry point [failed]
[HH:mm:ss] → Failed to resolve entry from: <rootDir>/test/__fixtures__/errors/invalid-entry.
`;

exports[`invalid-manifest 1`] = `
[HH:mm:ss] Resolving entry point [started]
[HH:mm:ss] → Resolving from: <rootDir>/test/__fixtures__/errors/invalid-manifest/src
[HH:mm:ss] → Resolved: <rootDir>/test/__fixtures__/errors/invalid-manifest/src/index.ts
[HH:mm:ss] Resolving entry point [completed]
[HH:mm:ss] Creating GitHub action bundle [started]
[HH:mm:ss] Creating GitHub action bundle [skipped]
[HH:mm:ss] → Action not found
[HH:mm:ss] Creating Node package bundles [started]
[HH:mm:ss] Creating Node package bundles [failed]
[HH:mm:ss] → Manifest file (package.json or action.yml) not found.
`;
